client/network/request-coalescer.ts

```diff
@@ -0,0 +1,32 @@
+import { BatchedAbortSignals } from '../../common/async/abort-signals'
+
+/**
+ * A helper class which allows you to use `BatchedAbortSignals` more easily. See the description
```

Comment:
Please make a comment that stands on its own, this class is for coalescing requests that take AbortSignals, the BatchedAbortSignals piece is an implementation detail and not something the consumer of this API needs to worry about